src/NuGet.Clients/NuGet.PackageManagement.UI/Utility/WpfUtility.cs

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Windows;
+using System.Windows.Media;
+
+namespace NuGet.PackageManagement.UI
+{
+    public static class WpfUtility
+    {
+        public static T FindParent<T>(this DependencyObject element)
```

Comment:
This is a nice utility! What is an example where we have an unknown tree structure in PMUI that can benefit from this? With your PR, is there an opportunity for an arbitrary tree structure where we would need this casting loop? So far, there's the AutomationPeer I made and one other place in PMUI where we explicitly walk the tree because it has a fixed hierarchy.

Reply:
This method is less about walking unknown tree structures (although it could be used for that), and more about removing code verbosity and duplication. For this PR, it is used to find a ListBoxItem of an item that each have a different parental tree structure, but both ultimately are contained in a ListBoxItem. The FindParent method can be used for any tree walking where we want to find a parent of a given type and do not want to specify every parent along the way.

Reply:
Alright. @rrelyea at one point I attempted a similar utility, I can't find the PR now, but the feedback I got was it was better perf to just explicitly walk the tree 1/2/3 levels up. The types were well-known. 
I'll let your input guide which direction we go here.

Reply:
don't remember that PR either.

Having a utility to walk up to the next StackPanel in your parent hierarchy is fine to do.
This is similar to the RelativeSource markup extension using FindAncestor that you'd use in markup.

Reply:
I think I like the name 'FindAncestor' better for this method too. Unless there's any objection, I'll change the name.